gear-programs/vft-treasury/src/wasm/tests/utils_gclient.rs

```diff
@@ -0,0 +1,226 @@
+/* gclient utils */
+use gclient::EventProcessor;
+use gclient::{EventListener, GearApi, Result};
+use gear_core::ids::*;
+use sails_rs::{Decode, Encode};
+use sails_rs::{H160, U256};
+use vft_treasury_app::services::{Config, InitConfig};
+pub async fn common_upload_program(
+    client: &gclient::GearApi,
+    code: Vec<u8>,
+    payload: Vec<u8>,
+) -> gclient::Result<([u8; 32], [u8; 32])> {
+    let gas_limit = client
+        .calculate_upload_gas(None, code.clone(), payload.clone(), 0, true)
+        .await?
+        .min_limit;
+    println!("init gas {gas_limit:?}");
+    let (message_id, program_id, _) = client
+        .upload_program_bytes(
+            code,
+            gclient::now_micros().to_le_bytes(),
+            payload,
+            gas_limit,
+            0,
+        )
+        .await?;
+
+    Ok((message_id.into(), program_id.into()))
+}
+
+pub async fn upload_program(
+    client: &gclient::GearApi,
+    listener: &mut gclient::EventListener,
+    code: &[u8],
+    payload: Vec<u8>,
+) -> gclient::Result<[u8; 32]> {
+    let (message_id, program_id) = common_upload_program(client, code.to_vec(), payload).await?;
+
+    assert!(listener
+        .message_processed(message_id.into())
+        .await?
+        .succeed());
+
+    Ok(program_id)
+}
+
+#[macro_export]
+macro_rules! send_request {
```

Comment:
Macro seems excess here. I suggest re-write it as function or remove it completely